### PR TITLE
feat: add kubernetes connection in exec connections

### PR DESCRIPTION
**Pull request by adityathebe**

### CODE REVIEW

File: connection/environment.go [hunk 1]
```diff
@@ -55,20 +58,43 @@ aws_secret_access_key = {{.SecretKey.ValueStatic}}
...
-func SetupConnection(ctx context.Context, connections ExecConnections, cmd *osExec.Cmd) error {
+func SetupConnection(ctx context.Context, connections ExecConnections, cmd *osExec.Cmd) (func() error, error) {
+	var cleaner = func() error {
+		return nil
+	}
+
+	if connections.Kubernetes != nil {
+		configPath, err := saveConfig(kubernetesConfigTemplate, connections.Kubernetes)
+		if err != nil {
+			return nil, fmt.Errorf("failed to store AWS credentials: %w", err)
```
Comment (moshloop) [inline connection/environment.go:75]:
```suggestion
			return nil, fmt.Errorf("failed to store kubernetes credentials: %w", err)
```

File: connection/environment.go [hunk 2]
```diff
@@ -9,15 +9,17 @@ import (
...
 
 	textTemplate "text/template"
 )
 
 // +kubebuilder:object:generate=true
 type ExecConnections struct {
-	AWS   *AWSConnection   `yaml:"aws,omitempty" json:"aws,omitempty"`
-	GCP   *GCPConnection   `yaml:"gcp,omitempty" json:"gcp,omitempty"`
-	Azure *AzureConnection `yaml:"azure,omitempty" json:"azure,omitempty"`
+	Kubernetes *types.EnvVar    `yaml:"kubernetes,omitempty" json:"kubernetes,omitempty"`
```
Comment (moshloop) [inline connection/environment.go:19]: Might make more sense as a KubernetesConnection object with a kubeconfig: EnvVar - in the future we might have 

```yaml
connections:
   kubernetes:
      eks:
         region: us-east-1
         clusterName: aws-demo
         connection: "connection://default/aws-eks-role"
```

File: connection/environment.go [hunk 3]
```diff
@@ -55,20 +63,73 @@ aws_secret_access_key = {{.SecretKey.ValueStatic}}
...
+		return nil
+	}
+
+	if connections.FromConfigItem != "" {
+		var configItem models.ConfigItem
+		if err := ctx.DB().Where("id = ?", connections.FromConfigItem).First(&configItem).Error; err != nil {
+			return nil, fmt.Errorf("failed to get config (%s): %w", connections.FromConfigItem, err)
+		}
+
+		if lo.FromPtr(configItem.Type) != "MissionControl::ScrapeConfig" {
```
Comment (moshloop) [inline connection/environment.go:83]: This can be any config_item, we just use the scraper id on it.

File: connection/kubernetes.go [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+		if err != nil {
+			return err
+		} else if connection == nil {
+			return fmt.Errorf("connection[%s] not found", t.ConnectionName)
+		}
+
+		t.KubeConfig.ValueStatic = connection.Certificate
+	}
+
+	if v, err := ctx.GetEnvValueFromCache(t.KubeConfig, ctx.GetNamespace()); err != nil {
```
Comment (moshloop) [inline connection/kubernetes.go:35]: The namespace of the playbook will unlikely match the namespace of the scrapeconfig

Comment (adityathebe) [reply connection/kubernetes.go:35]: namespace is set in the context in here

https://github.com/flanksource/duty/pull/1181/files#diff-e3e90426a3460155dcf682688699c1c9b2b2cb46ed156387b6efc1675808cfefR107

File: connection/environment.go [hunk 5]
```diff
@@ -55,20 +63,84 @@ aws_secret_access_key = {{.SecretKey.ValueStatic}}
...
+
+	if lo.FromPtr(connections.FromConfigItem) != "" {
+		var configItem models.ConfigItem
+		if err := ctx.DB().Where("id = ?", *connections.FromConfigItem).First(&configItem).Error; err != nil {
+			return nil, fmt.Errorf("failed to get config (%s): %w", *connections.FromConfigItem, err)
+		}
+
+		if lo.FromPtr(configItem.Type) != "MissionControl::ScrapeConfig" {
+			var scrapeConfig models.ConfigItem
+			if err := ctx.DB().Where("id = ?", lo.FromPtr(configItem.ScraperID)).First(&scrapeConfig).Error; err != nil {
```
Comment (moshloop) [inline connection/environment.go:85]: Why not just lookup scraper directly? We make not be using a CRD

File: connection/environment.go [hunk 6]
```diff
@@ -55,20 +65,90 @@ aws_secret_access_key = {{.SecretKey.ValueStatic}}
...
+			for _, kscraper := range kubernetesScrapers {
+				if kubeconfig, found, err := unstructured.NestedMap(kscraper.(map[string]any), "kubeconfig"); err != nil {
+					return nil, err
+				} else if found {
+					connections.Kubernetes = &KubernetesConnection{}
+					if err := runtime.DefaultUnstructuredConverter.FromUnstructured(kubeconfig, &connections.Kubernetes.KubeConfig); err != nil {
+						return nil, err
+					}
+
+					ctx = ctx.WithNamespace(scraperNamespace)
```
Comment (moshloop) [inline connection/environment.go:115]:
```suggestion
					connections.Kubernetes.Populate(ctx.WithNamespace(scraperNamespace))
```

**Pull request merged**